packages/odyssey-react-mui/src/labs/TopNav.tsx

```diff
@@ -0,0 +1,459 @@
+/*!
+ * Copyright (c) 2022-present, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import styled from "@emotion/styled";
+import { memo, useMemo, ReactElement, useCallback, KeyboardEvent } from "react";
+
+import type { HtmlProps } from "../HtmlProps";
+import {
+  AuraIcon,
+  AuraWordmarkIcon,
+  QuestionCircleIcon,
+  SettingsIcon,
+} from "../icons.generated";
+import { Link } from "../Link";
+import {
+  DesignTokens,
+  useOdysseyDesignTokens,
+} from "../OdysseyDesignTokensContext";
+
+export type TopNavLinkItem = {
+  id: string;
+  label: string;
+  /**
+   * link added to the nav item. if it is undefined, static text will be displayed.
+   * fires onClick event when it is passed
+   */
+  href?: string;
+  /**
+   * determines whether the link item is diabled
+   */
+  isDisabled?: boolean;
+  /**
+   * Event fired when the nav item is clicked
+   */
+  onClick?(): void;
```

Comment:
Minor thing, but this was hard for me to read because we hardly ever write it that way.

Since this isn't a `class` method, we should probably use the fat-arrow syntax:
```suggestion
  /**
   * Event fired when the nav item is clicked
   */
  onClick?: () => void;
```

But is there really no `event` we're passing through when we call their callback? That can be useful to consumers. We can get the type from React:

```suggestion
  /**
   * Event fired when the nav item is clicked
   */
  onClick?: MouseEventHandler<HTMLDivElement>;
```